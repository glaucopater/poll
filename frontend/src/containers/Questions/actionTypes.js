export default {
  QUESTIONS_LOADED: 'QUESTIONS_LOADED',
  VOTE_LOADED: 'VOTE_LOADED',
};
