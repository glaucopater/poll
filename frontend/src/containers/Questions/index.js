import React, { Fragment } from 'react';
import axios from 'axios';
import Choices from '../../components/Choices';
import EmptyResults from '../../components/EmptyResults';

import Moment from 'react-moment';

export default class Questions extends React.Component {
  constructor(props) {
    super(props);
    this.state = { questions: [] };
  }

  async fetchData() {
    axios
      .get(`/questions`, { crossDomain: true })
      .then(res => {
        if (res.data) {
          console.log(res.data);
          this.setState({ questions: res.data });
        }
      })
      .catch(function(error) {
        console.error('error:', error);
      });
  }

  async update() {
    await this.fetchData();
  }

  componentDidMount() {
    this.update();
  }

  render() {
    const questions = this.state.questions;
    return (
      <div className="Questions">
        {questions.length === 0 && <EmptyResults />}
        {questions.map((q, index) => (
          <Fragment key={index}>
            <span className="Question">{q.question}</span>
            <span className="Published_at">
              <Moment>{q.published_at}</Moment>
            </span>
            <span className="Url">{q.url}</span>
            <Choices {...q.choices} />
          </Fragment>
        ))}
      </div>
    );
  }
}
