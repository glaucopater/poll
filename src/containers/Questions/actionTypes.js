export default {
  QUESTIONS_LOADED: 'QUESTIONS_LOADED',
};
