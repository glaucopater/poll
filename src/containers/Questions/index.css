/*...App?*/
.Questions {
    padding: 2rem;
}

/*...Questions?*/
.Questions>ul {
    display: grid;
    grid-template-columns: repeat(auto-fit, minmax(320px, 1fr));
    grid-gap: 1rem;
}

/*Question*/
.Questions>ul>li {
    border: 1px solid #E2E2E2;
    border-radius: .5rem;
    list-style-type: none;
    box-shadow: 10px 5px 5px #00000030;
}

.Questions>ul>li>p {
    font-size: 1rem;
    line-height: 1.5;
    padding: 1rem .75rem;
    color: #666666;
}

.Questions>ul>li>a {

    padding: .5rem 1rem;
    margin: .5rem;
}