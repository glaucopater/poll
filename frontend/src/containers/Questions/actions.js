import actionTypes from './actionTypes.js';
import axios from 'axios';
import { QUESTIONS_ENDPOINT } from '../../constants';

export function fetchQuestions() {
  return dispatch => {
    return axios
      .get(QUESTIONS_ENDPOINT, { crossDomain: true })
      .then(resp => {
        dispatch({
          type: actionTypes.QUESTIONS_LOADED,
          questions: resp.data,
        });
      })
      .catch(error => {
        console.error('error:', error);
      });
  };
}

export function voteQuestion(choice) {
  return dispatch => {
    return axios
      .post(choice.url, { crossDomain: true })
      .then(resp => {
        dispatch({
          type: actionTypes.VOTE_LOADED,
          data: resp.data,
        });
      })
      .catch(error => {
        console.error('error:', error);
      });
  };
}
