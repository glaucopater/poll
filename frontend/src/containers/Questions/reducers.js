import actionTypes from './actionTypes';

const initState = {
  data: null,
};
export function fetchQuestionsReducer(state = initState, action) {
  switch (action.type) {
    case actionTypes.QUESTIONS_LOADED: {
      return {
        data: action.questions,
      };
    }
    default:
      return state;
  }
}
