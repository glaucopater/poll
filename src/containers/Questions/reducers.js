import actionTypes from './actionTypes';

const initState = {
  data: null,
};
export function fetchQuestionsReducer(state = initState, action) {
  switch (action.type) {
    case actionTypes.QUESTIONS_LOADED: {
      return {
        data: action.questions,
      };
    }
    default:
      return state;
  }
}

export function voteQuestionReducer(state = initState, action) {
  switch (action.type) {
    case actionTypes.VOTE_LOADED: {
      return {
        data: action.data,
      };
    }
    default:
      return state;
  }
}
