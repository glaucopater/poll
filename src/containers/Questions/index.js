import React from 'react';
import PropTypes from 'prop-types';
import Question from '../../components/Question';
import EmptyResults from '../../components/EmptyResults';
import { fetchQuestions } from './actions';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import Strings from '../../constants/strings';

import { StyledQuestions, StyledQuestionsGrid } from './styled';
class Questions extends React.PureComponent {
  static propTypes = {
    questions: PropTypes.object,
    fetchQuestions: PropTypes.func,
  };
  constructor(props) {
    super(props);
    this.state = { questions: [] };
  }

  async update() {
    this.props.fetchQuestions();
  }

  componentDidMount() {
    this.update();
  }

  render() {
    const questions =
      this.props.questions && this.props.questions.data
        ? this.props.questions.data
        : [];

    return (
      <StyledQuestions>
        {questions.length === 0 && <EmptyResults />}
        {questions.length > 0 && <h1>{Strings.questions}</h1>}
        <StyledQuestionsGrid>
          {questions.map((q, index) => (
            <Question key={index} {...q} />
          ))}
        </StyledQuestionsGrid>
      </StyledQuestions>
    );
  }
}

export default connect(
  ({ fetchQuestions: questions, voteQuestion: data }) => ({
    questions,
    data,
  }),
  dispatch =>
    bindActionCreators(
      {
        fetchQuestions,
      },
      dispatch,
    ),
)(Questions);
