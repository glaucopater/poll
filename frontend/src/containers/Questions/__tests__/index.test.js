import { shallow } from 'enzyme';
import React from 'react';
import Questions from '../index';

describe('<Questions/>', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Questions />);
    expect(wrapper).toMatchSnapshot();
  });
});
