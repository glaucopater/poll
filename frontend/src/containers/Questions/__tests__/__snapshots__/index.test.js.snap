// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Questions/> renders correctly 1`] = `ShallowWrapper {}`;
